CMakeLists.txt

```diff
@@ -17,6 +17,7 @@ option(DISABLE_DOWNLOADS "Disable downloads of dependencies during build." OFF)
 option(DOWNLOAD_ENSMALLEN "If ensmallen is not found, download it." ON)
 option(DOWNLOAD_STB_IMAGE "Download stb_image for image loading." ON)
 option(BUILD_GO_SHLIB "Build Go shared library." OFF)
+option(BUILD_DOCS "Build doxygen documentation (if doxygen is available)." ON)
```

Comment:
Wondering if the default should be OFF, or if not if we should set `WARN_AS_ERROR          = NO` as default?

Reply:
Yeah, good point, let's set `WARN_AS_ERROR` to `NO` by default, but then sed it to `ON` in the Doxygen build.

Reply:
Done in 5552b8caa.

Reply:
Sounds like a good solution to me.